src/api/__tests__/hooks.unit.test.ts.tsx

```diff
@@ -1,533 +1,315 @@
 
-    const [resp2, err2] = result.current;
-    expect(resp2.next).not.toBeNull();
-    expect(resp2.previous).not.toBeNull();
-    expect(resp2.page).toEqual(data);
-    expect(err2).toBeUndefined();
-
-    act(() => {
-      resp2.previous!();
-    });
-
-    expect(mockAxios.get.mock.calls[2][0]).toBe("test2");
+    [resp, error, info] = result.current;
+    expect(info.state).toBe(RequestState.Complete);
   });
 
-  it("returns a null next paginator if next page does not exist", () => {
-    const { result } = renderHook(() => h.useApiGetPaginated("test"));
-    const data = {
-      count: 20,
-      page: 2,
-      num_pages: 2,
-      next: null,
-      previous: "test",
-      results: []
-    };
-
-    act(() => {
-      mockAxios.mockResponse({
-        data: {
-          success: true,
-          data
-        }
-      });
+  it("correctly updates return value on multiple fetches", async () => {
+    const get = retrieveFactory<string>("bingo/");
+    const { result, rerender } = renderHook(({ fn }) => useAPI(fn), {
+      initialProps: {
+        fn: () => get({ id: "15" }),
+      },
     });
 
-    const [{ next, previous }] = result.current;
-
-    expect(next).toBeNull();
-    expect(typeof previous).toBe("function");
-  });
-
-  it("returns a null previous paginator if previous page does not exist", () => {
-    const { result } = renderHook(() => h.useApiGetPaginated("test"));
-    const data = {
-      count: 20,
-      page: 2,
-      num_pages: 2,
-      next: "test",
-      previous: null,
-      results: []
-    };
-
-    act(() => {
-      mockAxios.mockResponse({
-        data: {
-          success: true,
-          data
+    await act(async () => {
+      mockAxios.mockResponseFor(
+        {
+          method: "GET",
+          url: "bingo/15/",
+        },
+        {
+          data: MOCK_ERROR,
         }
-      });
+      );
     });
 
-    const [{ next, previous }] = result.current;
+    let [resp, error] = result.current;
 
-    expect(typeof next).toBe("function");
-    expect(previous).toBeNull();
-  });
+    expect(resp).toBeUndefined();
+    expect(error).toEqual(MOCK_ERROR.error);
 
-  it("does not crash on an error", () => {
-    const { result } = renderHook(() => h.useApiGetPaginated("test"));
+    // create a new function to get the hook to re run the request
+    const fn2 = () => get({ id: "15" });
+    rerender({ fn: fn2 });
 
-    act(() => {
-      mockAxios.mockError(makeTestError(500, ERRORS.__TESTING));
+    await act(async () => {
+      mockAxios.mockResponseFor(
+        {
+          method: "GET",
+          url: "bingo/15/",
+        },
+        {
+          data: MOCK_RESPONSE,
+        }
+      );
     });
 
-    const [, err] = result.current;
+    [resp, error] = result.current;
 
-    expect(err).toBe(ERRORS.__TESTING);
+    expect(resp).toEqual(MOCK_RESPONSE.data);
+    expect(error).toBeUndefined();
   });
 });
 
-describe("useApiPost", () => {
+describe("useAPILazy", () => {
```

Comment:
Missing test cases for unsuccessful requests

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/33/commits/9522bc49a1895c48b860f58aa1c765cbf918e6ad